tests/src/Functional/NodeFormTest.php

```diff
@@ -87,9 +87,32 @@ public function testNodeForm() {
     $this->drupalGet('node/add/page');
     $web_assert->optionExists(WorkbenchAccessManagerInterface::FIELD_NAME, $staff_term->getName());
 
+    // Test that we cannot use autocomplete to save a term we cannot access.
+    $this->setFieldType('node', 'page', 'entity_reference_autocomplete');
+
+    $this->drupalGet('node/add/page');
+    $field = WorkbenchAccessManagerInterface::FIELD_NAME . "[0][target_id]";
+
+    // Try to save something that doesn't exist.
+    $this->submitForm([$field => 'garbage', 'title[0][value]' => 'Foo'], 'Save');
+    $web_assert->pageTextContains('There are no entities matching "garbage".');
+
+    // Try to force an invalid selection.
+    $this->submitForm([$field => $super_staff_term->label() . ' (' . $super_staff_term->id() . ')', 'title[0][value]' => 'Foo'], 'Save');
+    $web_assert->pageTextContains('The referenced entity (taxonomy_term: 2) does not exist.');
```

Comment:
I'm not sure offhand how we would override that string

Reply:
It is coming from core. In Drupal 7, we would reset the messages in drupal_set_message(). Here, in Drupal 8, that's OO code.

Get current messages:

https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Messenger%21MessengerInterface.php/function/MessengerInterface%3A%3AmessagesByType/8.5.x

Inside that class yuou can read and then empty the messages array.

Then use:

https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Messenger%21MessengerInterface.php/function/MessengerInterface%3A%3AaddMessage/8.5.x

To add the messages we wish to keep and our rewritten message.

I think this can wait for a follow-up.